### PR TITLE
Change BufferedHtmlContent to treat strings as unencoded

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

**Comment by rynowak** [comment]
Reaction PRs in MVC and Razor to follow.

File: NuGet.Config [hunk 1]
```diff
@@ -1,7 +1,10 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
     <add key="AspNetVNext" value="https://www.myget.org/F/aspnetlitedev/api/v2" />
     <add key="NuGet.org" value="https://api.nuget.org/v3/index.json" />
   </packageSources>
+  <disabledPackageSources>
+    <add key="nuget.org" value="true" />
+  </disabledPackageSources>
```
Comment (rynowak) [inline NuGet.Config]: will undo

File: Common.sln [hunk 2]
```diff
@@ -95,7 +94,6 @@ Global
 		{0BF8866B-A179-4A6E-AAFD-BE487664298B} = {FEAA3936-5906-4383-B750-F07FE1B156C5}
 		{01770CF7-74F5-4257-BF1E-3F5674B413CF} = {FEAA3936-5906-4383-B750-F07FE1B156C5}
 		{31D4C34C-436F-48E9-B8E4-641574741887} = {FEAA3936-5906-4383-B750-F07FE1B156C5}
-		{B4B564DE-3066-4A31-97F6-077461CDA454} = {FEAA3936-5906-4383-B750-F07FE1B156C5}
```
Comment (rynowak) [inline Common.sln]: will not undo. Not going to fight vs on this one.

File: test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs [hunk 3]
```diff
@@ -131,20 +150,22 @@ public void WriteTo_WritesAllItems()
 
             // Assert
             Assert.Equal(2, content.Entries.Count);
-            Assert.Equal("Written from TestHtmlContent: HelloTest", writer.ToString());
+            Assert.Equal("Written from TestHtmlContent: HelloHtmlEncode[[Test]]", writer.ToString());
         }
 
+        // We're purposely avoiding anything here that actually gets encoded. We don't want to take a dependency
+        // on the actual encoder implementation.
```
Comment (dougbu) [inline test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs]: Why not use `CommonTestEncoder`? This test project already references the `Microsoft.Framework.WebEncoders.Testing` package.

File: src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs [hunk 4]
```diff
@@ -147,5 +147,16 @@ public override string ToString()
                 return writer.ToString();
             }
         }
+
+        // Avoid encoding newline characters
+        private class NewLine : IHtmlContent
```
Comment (dougbu) [inline src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs]: Suggest this class should implement `AppendEncoded(string)` using another `private` class. We're going that route (i.e. that method should be present in `IHtmlContentBuilder`) and the value of moving `HtmlString` some place central seems very low.

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs (hunk 3 above)
Comment (rynowak) [inline test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs]: This test is for the `ToString` method. I was about 50/50 before on just removing the ToString() and replacing it with a debugger display. Now I feel pretty strongly that removing ToString and this test is the right call. Thoughts?

File: src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs (hunk 4 above)
Comment (rynowak) [inline src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs]: Sure

File: test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs (hunk 3 above)
Comment (dougbu) [inline test/Microsoft.Framework.Internal.Test/BufferedHtmlContentTest.cs]: :+1:

**Comment by rynowak** [comment]
@dougbu Updated

**Comment by dougbu** [comment]
:shipit: once the reactions are ready. Do Entropy or MusicStore also need reaction PRs?

**Pull request closed without merging**